test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,370 @@
+  EXPECT_EQ(getStreamField(data_message_1, "rollingCountSuccess"), "70");
+  EXPECT_EQ(getStreamField(data_message_1, "rollingCountTimeout"), "30");
+  EXPECT_EQ(getStreamField(data_message_1, "errorPercentage"), "79");
+
+  // check stream format and data
+  EXPECT_EQ(getStreamField(data_message_2, "errorCount"), "70"); // note that on regular operation,
+                                                                 // 5xx and timeout are raised
+                                                                 // together, so timeouts are
+                                                                 // reduced from 5xx count
+  EXPECT_EQ(getStreamField(data_message_2, "requestCount"), "190");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSemaphoreRejected"), "50");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSuccess"), "30");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountTimeout"), "40");
+  EXPECT_EQ(getStreamField(data_message_2, "errorPercentage"), "84");
+
+  buffer.drain(buffer.length());
+
+  // removing cluster
+  cluster_map_.erase(cluster2_name);
+  ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+  sink_->flush(source_);
+  data_message = TestUtility::bufferToString(buffer);
+
+  EXPECT_NE(std::string::npos, data_message.find(cluster1_name_));
+  EXPECT_EQ(std::string::npos, data_message.find(cluster2_name));
+
+  // add cluster again
+  buffer.drain(buffer.length());
+  cluster_map_.emplace(cluster2_name, cluster2);
+  ON_CALL(error_5xx_counter2, value()).WillByDefault(Return(0));
+  ON_CALL(retry_5xx_counter2, value()).WillByDefault(Return(0));
+  ON_CALL(error_4xx_counter2, value()).WillByDefault(Return(0));
+  ON_CALL(retry_4xx_counter2, value()).WillByDefault(Return(0));
+  ON_CALL(success_counter2, value()).WillByDefault(Return(0));
+  cluster2_info->stats().upstream_rq_timeout_.reset();
+  cluster2_info->stats().upstream_rq_per_try_timeout_.reset();
+  cluster2_info->stats().upstream_rq_pending_overflow_.reset();
+  ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+  sink_->flush(source_);
+
+  data_message = TestUtility::bufferToString(buffer);
+  pos1 = data_message.find(cluster1_name_);
+  pos2 = data_message.find(cluster2_name);
+  EXPECT_NE(std::string::npos, pos1);
+  EXPECT_NE(std::string::npos, pos2);
+
+  data_message_2 = (pos2 < pos1) ? data_message.substr(pos2, pos1) : data_message.substr(pos2);
+
+  // check that old values of test_cluster2 were deleted
+  EXPECT_EQ(getStreamField(data_message_2, "errorPercentage"), "0");
+  EXPECT_EQ(getStreamField(data_message_2, "errorCount"), "0");
+  EXPECT_EQ(getStreamField(data_message_2, "requestCount"), "0");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSemaphoreRejected"), "0");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSuccess"), "0");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountTimeout"), "0");
```

Comment:
My general comment here is that I can only peephole review the tests because they are so long, so it's hard for me to tell what's going on.  Can you factor out some common blocks into helpers?  E.g. this pattern of checking that a specific bunch of values in a message are 0 is repeated 3x in this file.  Factor that out into a common method.  This is also the case likely with a bunch of other stanzas, such as as setting up a bunch of mocks.